### PR TITLE
confirm destination URL before open link

### PR DESCRIPTION
**Pull request by hell0-Wor1d**
Show the real destination URL to avoid phishing attempt.

#187

### CODE REVIEW

File: src/app.ts [hunk 1]
```diff
@@ -305,6 +306,12 @@ app.on('before-quit', () => {
       event.newGuest = win
     }
 
-    shell.openExternal(cleanURLFromGoogle(url))
+    dialog.showMessageBox({
+      type: 'info',
+      buttons: ['Yes', 'No'],
```
Comment (timche) [inline src/app.ts:311]:
```suggestion
      buttons: ['Open Link', 'Cancel'],
```

File: src/app.ts [hunk 2]
```diff
@@ -305,6 +306,30 @@ app.on('before-quit', () => {
...
+    }
+
+    dialog
+      .showMessageBox({
+        type: 'info',
+        buttons: ['Open Link', 'Cancel'],
+        title: `Do you want to open the external website "${cleanURL}" in your default browser?`,
+        checkboxLabel: `Trust all links on ${origin}`,
+        message: cleanURL
+      })
```
Comment (timche) [inline src/app.ts:325]:
```suggestion
    dialog
      .showMessageBox({
        type: 'info',
        buttons: ['Open Link', 'Cancel'],
        message: `Do you want to open the external website "${cleanURL}" in your default browser?`,
        checkboxLabel: `Trust all links on ${origin}`,
        detail: cleanURL
      })
```

**Comment by timche** [comment]
There are also other places where external URLs are opened:
- https://github.com/timche/gmail-desktop/blob/master/src/app.ts#L301
- https://github.com/timche/gmail-desktop/blob/master/src/app.ts#L308

Could you please wrap the stuff in a function and re-use that in these places too?

**Comment by hell0-Wor1d** [comment]
Done !
It look like this on Windows : 
Capture

File: src/app.ts [hunk 3]
```diff
@@ -168,6 +169,10 @@ function addCustomCSS(windowElement: BrowserWindow): void {
   }
 }
 
+function openExternalUrl(url: string): void {
+  shell.openExternal(url)
+}
+
```
Comment (timche) [inline src/app.ts:175]: 😂 

You understood me wrong.

I meant you should wrap L313-L337 into a function, so everywhere where an external URL can be opened, should have the same behavior with trusting URLs.

**Comment by timche** [comment]
Awesome, cheers for your work 🙇

**Comment by timche** [comment]
@all-contributors please add @hell0-Wor1d for code

**Pull request merged**